src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -2209,5 +2209,13 @@ public SyntaxNode LambdaParameter(string identifier, ITypeSymbol type)
         public abstract SyntaxNode TupleExpression(IEnumerable<SyntaxNode> arguments);
 
         #endregion
+
+        #region Patterns
+
+        internal abstract bool SupportsPatterns(ParseOptions options);
+        internal abstract SyntaxNode IsPatternExpression(SyntaxNode expression, SyntaxNode pattern);
```

Comment:
Nit: The name is unfortunate. Sounds like a method to test for something `Is...`.

Reply:
Unfortunate, but consistent with the rest of the api...